src/shared/firewall-util-nft.c

```diff
@@ -423,19 +428,31 @@ static int sd_nfnl_message_new_dnat_rule_out(sd_netlink *nfnl, sd_netlink_messag
         if (r < 0)
                 return r;
 
-        /* 1st statement: exclude 127.0.0.1/8: ip daddr != 127.0.0.1/8 */
-        r = nfnl_add_expr_payload(m, NFT_PAYLOAD_NETWORK_HEADER, offsetof(struct iphdr, daddr),
-                                  sizeof(uint32_t), NFT_REG32_01);
-        if (r < 0)
-                return r;
+        /* 1st statement: exclude 127.0.0.1/8: ip daddr != 127.0.0.1/8, resp. avoid ::1 */
+        if (family == AF_INET) {
+                uint32_t lonet = htobe32(0x7F000000), lomask = htobe32(0xff000000);
```

Comment:
btw, the lomask definition is a bit weird.

0xff000000 is actually an expression that the C compiler resolves as signed integer (since you did not specify a suffix), but it doesn't fit into a signed integer. You then pass it to a function that expectes an unsigned integer, which then fixes things up again. It's a bit ugly though, since you wrote something that looks like an unsigned is used like an unsigned but as actually signed.

i.e. please wrap the constant in UINT32_C(…), i.e. the offical C99 way to encode 32bit unsigned constants with the right suffix